include/seqan3/alignment/scoring/detail/simd_matrix_scoring_scheme.hpp

```diff
@@ -0,0 +1,145 @@
+    constexpr simd_matrix_scoring_scheme(simd_matrix_scoring_scheme &&) = default; //!< Defaulted.
+    constexpr simd_matrix_scoring_scheme & operator=(simd_matrix_scoring_scheme const &) = default; //!< Defaulted.
+    constexpr simd_matrix_scoring_scheme & operator=(simd_matrix_scoring_scheme &&) = default; //!< Defaulted.
+    ~simd_matrix_scoring_scheme() = default; //!< Defaulted.
+
+    //!\copydoc seqan3::detail::simd_matrix_scoring_scheme::initialise_from_scalar_scoring_scheme
+    constexpr explicit simd_matrix_scoring_scheme(scoring_scheme_t const & scoring_scheme)
+    {
+        initialise_from_scalar_scoring_scheme(scoring_scheme);
+    }
+
+    //!\copydoc seqan3::detail::simd_matrix_scoring_scheme::initialise_from_scalar_scoring_scheme
+    constexpr simd_matrix_scoring_scheme & operator=(scoring_scheme_t const & scoring_scheme)
+    {
+        initialise_from_scalar_scoring_scheme(scoring_scheme);
+        return *this;
+    }
+    //!\}
+
+    /*!\name Score computation
+     *\{
+     */
+    /*!\brief Given two simd vectors, compute an element-wise score and return in another simd vector.
+     * \param[in] lhs The left operand to compare.
+     * \param[in] rhs The right operand to compare.
+     * \returns A simd vector of same type as input, with scores computed between both inputs.
+     *
+     * ### Exception
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Linear in the length of one input vector. (`lhs` and `rhs` are equally sized)
+     *
+     * ### Thread safety
+     *
+     * Thread-safe.
+     */
+    constexpr simd_score_t score(simd_score_t const & lhs, simd_score_t const & rhs) const noexcept
+    {
+        simd_score_t result;
+
+        for (int i = 0; i < simd_traits<simd_score_t>::length; i++)
+        {
+            if (is_padded(lhs[i]) || is_padded(rhs[i]))
+            {
+                result[i] = std::same_as<alignment_t, detail::global_alignment_type> ? 1 : -1;
+            }
+            else
+            {
+                result[i] = internal_scoring_scheme.score(assign_rank_to(lhs[i], alphabet_t{}),
+                                                          assign_rank_to(rhs[i], alphabet_t{}));
+            }
+        }
+
+        return result;
+    }
+    //!\}
+
+private:
+    //!\brief Internally stores the given scalar scoring scheme matrix.
+    scoring_scheme_t internal_scoring_scheme;
+
+    /*!\brief Store the given scoring scheme matrix into a private member variable.
+     * \param[in] scoring_scheme The scoring scheme to initialise the vectorised match and mismatch score with.
+     */
+    constexpr void initialise_from_scalar_scoring_scheme(scoring_scheme_t const & scoring_scheme)
+    {
+        internal_scoring_scheme = scoring_scheme;
+    }
+
+    /*!\brief Checks whether the given value is a padded value or not. This check is done by casting the scalar
```

Comment:
Everything after the first sentence is not brief anymore but should go into \details of this documentation